File: crypto/src/main/java/io/oversec/one/crypto/encoding/pad/ManualPadder.java
package io.oversec.one.crypto.encoding.pad;

import android.content.Context;
import io.oversec.one.crypto.R;

public class ManualPadder extends AbstractPadder {
    ManualPadder(Context ctx) {
        super(ctx);
    }

    @Override
    public synchronized void pad(String orig, StringBuffer encoded) {
    }

    @Override
    public void reset() {
    }

    @Override
    public char getNextPaddingChar() {
        return 0;
    }

    @Override
    String tail() {
        return "";
    }

    @Override
    public String getId() {
        return mCtx.getString(R.string.padder_manual);
    }

    @Override
    public String getLabel() {
        return mCtx.getString(R.string.padder_manual);
    }


    @Override
    public String getExample() {
        return mCtx.getString(R.string.manual_padder_hint);
    }
}

File: crypto/src/main/java/io/oversec/one/crypto/encoding/pad/ManualPadder.kt
package io.oversec.one.crypto.encoding.pad

import android.content.Context
import io.oversec.one.crypto.R

class ManualPadder internal constructor(ctx: Context) : AbstractPadder(ctx) {

    @Synchronized
    override fun pad(orig: String, encoded: StringBuffer) {
    }

    override fun reset() {}

    override val nextPaddingChar: Char
        get() = 0.toChar()

    override fun tail(): String {
        return ""
    }

    override val id
        get() = mCtx.getString(R.string.padder_manual)

    override val label
        get() = mCtx.getString(R.string.padder_manual)

    override val example
        get() = mCtx.getString(R.string.manual_padder_hint)

}
